### ISSUE TITLE
[a11y]: Accessibility checker finds violation on carbon tooltip

### ISSUE THREAD

**Issue by miriam-g** [carbon-design-system/carbon#18921]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

carbon/react 1.78.0

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest deployment

### Description

We have a checkbox with label and a tooltip on this label.
This leads to the following accessibility violation found by the IBM Accessibility Checker:

> The ARIA attributes "aria-labelledby" are not valid for the element <span> with implicit ARIA role "generic" [Learn more](https://able.ibm.com/rules/archives/2025.02.10/doc/en-US/aria_attribute_valid.html#%7B%22message%22%3A%22The%20ARIA%20attributes%20%5C%22aria-labelledby%5C%22%20are%20not%20valid%20for%20the%20element%20%3Cspan%3E%20with%20implicit%20ARIA%20role%20%5C%22generic%5C%22%22%2C%22snippet%22%3A%22%3Cspan%20aria-labelledby%3D%5C%22tooltip-318%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_invalid_implicit_role_attr%22%2C%22ruleId%22%3A%22aria_attribute_valid%22%2C%22msgArgs%22%3A%5B%22aria-labelledby%22%2C%22span%22%2C%22generic%22%5D%7D)

### Reproduction/example

https://stackblitz.com/edit/github-wuax8rlb?file=src%2FApp.jsx&preset=node=

### Steps to reproduce

View the code in stackblitz to reproduce the issue

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
When you use the label prop in the Tooltip component, it applies an `aria-labelledby `attribute to whatever it wraps. In this case, it's applying it to a `span` element, which has an implicit role of "generic" that doesn't support this attribute.

to fix the warning - Use the `description` prop instead of label in the Tooltip, this will cause the Tooltip to use `aria-describedby` instead of `aria-labelledby`, which is more appropriate for a span element and will resolve the accessibility warning
here is the [stackblitz](https://stackblitz.com/edit/github-wuax8rlb-q12chepe?file=src%2FApp.jsx&preset=node=)
